### DISCUSSION TITLE
Trying to integrate joltphysics into a custom game engine, need help

### DISCUSSION THREAD

**Discussion by bbq-porkchop** [jrouwe/JoltPhysics#1288] [Q&A] [answered]
Followed HelloWorld.cpp, however updating the physics system causes a crash with read access violation.
![Screenshot 2024-09-30 094820](https://github.com/user-attachments/assets/0f26f955-ba93-4cb2-9939-98e3d8ccdc05)
![Screenshot 2024-09-30 094544](https://github.com/user-attachments/assets/7e19c317-92a4-498a-a828-11a076470ab2)

**Comment by jrouwe** [comment, marked as answer]
The interfaces you pass to Physics System::Init have been destructed by the time the physics system updates (because they are locals to your init function). They need the same lifetime as the PhysicsSystem.

**Reply by bbq-porkchop** [reply]
Thank you!
